src/Graphing.hs

```diff
@@ -61,6 +62,14 @@ filter f gr = gr { graphingDirect = direct', graphingAdjacent = adjacent' }
 empty :: Graphing ty
 empty = Graphing S.empty AM.empty
 
+-- | Strip a given item from the direct set, promote its immediate children to direct items
+stripRoot :: Ord ty => Graphing ty -> Graphing ty
+stripRoot gr = gr { graphingDirect = direct' }
+  where
+  topLayer = graphingDirect gr
+  newDirects = S.unions $ map (flip AM.postSet $ graphingAdjacent gr) $ S.toList topLayer
+  direct' = foldr S.insert S.empty newDirects
```

Comment:
I think `direct'` is going to have the same value as `newDirects` -- `newDirects` is already `:: Set ty`. good intuition on the foldr here in general, though 🙂

Reply:
Ah yeah.  That came from my first attempt, which was based on removing a named item from `directs`.